src/components/ImportAccountsDrawer/ImportAccountsDrawer.tsx

```diff
@@ -0,0 +1,249 @@
+  // which internally calls dispatch(portfolio.actions.upsertPortfolio(account)) and upserts assets
+  // But also need to remove accounts that were disabled.
+
+  // initial fetch to detect the number of accounts based on the "first empty account" heuristic
+  useEffect(() => {
+    ;(async () => {
+      if (!wallet) return
+
+      let accountNumber = 0
+      const accounts = []
+
+      while (true) {
+        try {
+          const accountResult = await queryClient.fetchQuery(
+            reactQueries.accountManagement.loadAccount(accountNumber, chainId, wallet),
+          )
+
+          if (!accountResult) break
+
+          const { accountId, hasActivity } = accountResult
+
+          // Add the account before checking if it has activity so user has ability to toggle it if needed.
+          accounts.push({ accountNumber, accountId })
+
+          // If the account has no activity, it's the first empty account.
+          if (!hasActivity) {
+            break
+          }
+        } catch (error) {
+          console.error(error)
+          break
+        }
+
+        accountNumber++
+      }
+
+      setAccounts(accounts)
+    })()
+  }, [chainId, queryClient, wallet])
+
+  const handleLoadMore = useCallback(async () => {
+    if (!wallet) return
+    const accountNumber = accounts.length
+    const accountResult = await queryClient.fetchQuery(
+      reactQueries.accountManagement.loadAccount(accountNumber, chainId, wallet),
+    )
+    if (!accountResult) return
+    setAccounts(previousAccounts => {
+      const { accountId } = accountResult
+      return [...previousAccounts, { accountNumber, accountId }]
+    })
+  }, [accounts, chainId, queryClient, wallet])
+
+  const handleToggleAccountId = useCallback((accountId: AccountId) => {
+    console.log('handleToggleAccountId', accountId)
+  }, [])
+
+  const accountRows = useMemo(() => {
+    if (!feeAsset) return null
+    return accounts.map(({ accountId, accountNumber }) => (
+      <TableRow
+        key={accountId}
+        accountId={accountId}
+        accountNumber={accountNumber}
+        feeAsset={feeAsset}
+        toggleAccountId={handleToggleAccountId}
+      />
+    ))
+  }, [accounts, feeAsset, handleToggleAccountId])
+
+  if (!feeAsset) {
+    console.error(`No fee asset found for chainId: ${chainId}`)
+    return null
+  }
+
+  return (
+    <Drawer isOpen={isOpen} size='lg' placement='right' onClose={onClose}>
+      <DrawerOverlay />
+      <DrawerContent>
+        <DrawerCloseButton />
+        <DrawerHeader>
+          <RawText as='h3'>
+            {translate('accountManagementSidebar.title', { chainNamespaceDisplayName })}
+          </RawText>
+          <Text
+            color='text.subtle'
+            fontSize='md'
+            translation={'accountManagementSidebar.description'}
+          />
+        </DrawerHeader>
+
+        <DrawerBody>
+          <TableContainer>
```

Comment:
gm!

It might deserve a margin bottom so it's never sticky with other bottom elements

Also don't we have a component somewhere for the load more button? It feels a bit weird as it is and would deserve a bit more of styling from my point of view, I'm not so used of the actual UI so I might be wrong

Reply:
![image](https://github.com/shapeshift/web/assets/14963751/10d65184-2d8b-4347-a70b-bfdaab482303)